code/controllers/subsystem/persistent_data.dm

```diff
@@ -0,0 +1,58 @@
+/**
+  * # Persistent Data Subsystem
+  *
+  * Provides a centralised handler for persistent data reading and writing.
+  * The subsystem does not do any actual spawning itself, as this focuses on objects and mobs
+  * Should anything that is turf persistence related added in, that can be chucked into this SS
+  * Its quite a simple subsystem. For now, anyways.
+  *
+  */
+SUBSYSTEM_DEF(persistent_data)
+	name = "Persistent Data"
+	init_order = INIT_ORDER_PERSISTENCE // -95 | Loads after EVERYTHING else
+	flags = SS_NO_FIRE
+	/// List of atoms registered into the subsystem for persistent data storage. Can be anything at all
+	var/list/registered_atoms = list()
+
+/datum/controller/subsystem/persistent_data/Initialize()
+	// Load all the data of registered atoms
+	for(var/atom/A in registered_atoms)
```

Comment:
Why only atoms? I can see myself using this for instance for the logging tools or other backend logic that is not managed by a SS.
Can also be made into a non typed loop if there is only one type in the list. Same as the one below

Reply:
I cant think of any datums which would have persistent storage, which wouldnt be externally managed (Example: When I do map rotation, the saving and loading of data will be handled by the mapping SS). It also relies on the atoms existing to be added to this list, as well as adding them with `Initialize()`. Instancing datums becomes an issue, and should be managed by their own service. If you wanted to do this for logging tools, it should be made into a subsystem, so all the vars can be associated with it, especially since I spent time before killing off orphaned controllers which were hooked into other subsystems in less than optimal ways.

Reply:
Isn't `registered_atoms` always empty on server start? I'm probably missing the part where data is carried over to the next round.

Reply:
`SSatoms` initialises before `SSpersistent_data`, which adds the atoms into the `SSatoms.registered_atoms` list.

Reply:
Ah ok, and `SSpersistent_data.register(src)` is called in `New` of Ian and runtime, which adds it to the list. I see.